modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/GridSqlQueryParser.java

```diff
@@ -1323,9 +1293,64 @@ private GridSqlCreateTable parseCreateTable(CreateTable createTbl) {
                 res.affinityKey(pkCols0.iterator().next());
         }
 
+        if (implicitPk) {
+            log.warning("No PRIMARY KEY defined for table '" + data.tableName + "'. " +
```

Comment:
nitpicking: at this point you don't know yet if an implicit pk will be created or this command will be rejected by another round of validation